cmd/registry/breaking-change-detector/breaking_change_detector.go

```diff
@@ -0,0 +1,212 @@
+package breakingchangedetector
+
+import (
+	"regexp"
+
+	"github.com/apigee/registry/rpc"
+)
+
+type detectionPattern struct {
+	detectionWeight    int
+	PositiveMatchRegex *regexp.Regexp
+	NegativeMatchRegex *regexp.Regexp
+}
+
+var (
+	unsafeAdds = []detectionPattern{
+		{
+			detectionWeight:    1,
+			PositiveMatchRegex: regexp.MustCompile("(components.)+(.|)+(schemas)+(.)+(required)"),
+		},
+	}
+
+	unsafeDeletes = []detectionPattern{
+		{
+			detectionWeight:    1,
+			PositiveMatchRegex: regexp.MustCompile("(components.)+(.|)+(schemas)+(.)"),
+			NegativeMatchRegex: regexp.MustCompile("(components.)+(.|)+(schemas)+(.)+(required)"),
+		},
+		{
+			detectionWeight:    1,
+			PositiveMatchRegex: regexp.MustCompile("(paths.)+(.|)"),
+			NegativeMatchRegex: regexp.MustCompile("((paths.)+(.|)+(tags))+(|)+((paths.)+(.|)+(description))"),
+		},
+	}
+
+	unsafeMods = []detectionPattern{
+		{
+			detectionWeight:    1,
+			PositiveMatchRegex: regexp.MustCompile("(components.)+(.|)+(schemas)+(.|)+(type)"),
+		},
+		{
+			detectionWeight:    1,
+			PositiveMatchRegex: regexp.MustCompile("(paths.)+(.|)+(type)"),
+			NegativeMatchRegex: regexp.MustCompile("((paths.)+(.|)+(tags))+(|)+((paths.)+(.|)+(description))"),
+		},
+	}
+
+	safeAdds = []detectionPattern{
+		{
+			detectionWeight:    1,
+			PositiveMatchRegex: regexp.MustCompile("(info.)+(.)"),
+		},
+		{
+			detectionWeight:    1,
+			PositiveMatchRegex: regexp.MustCompile("(tags.)+(.)"),
+		},
+	}
+
+	safeDeletes = []detectionPattern{
+		{
+			detectionWeight:    1,
+			PositiveMatchRegex: regexp.MustCompile("(info.)+(.)"),
+		},
+		{
+			detectionWeight:    1,
+			PositiveMatchRegex: regexp.MustCompile("(tags.)+(.)"),
+		},
+	}
+
+	safeMods = []detectionPattern{
+		{
+			detectionWeight:    1,
+			PositiveMatchRegex: regexp.MustCompile("(info.)+(.)"),
+		},
+		{
+			detectionWeight:    1,
+			PositiveMatchRegex: regexp.MustCompile("(tags.)+(.)"),
+		},
+	}
+)
+
+// GetBreakingChanges compares each change in a diff Proto to the relavant change type detection Patterns.
+// Each change is then catgorized as breaking, nonbreaking, or unknown.
+func GetBreakingChanges(diff *rpc.Diff) *rpc.ClassifiedChanges {
```

Comment:
Should this be `GetClassifiedChanges` now?

Reply:
Done